### ISSUE TITLE
TypeError: date.format is not a function

### ISSUE THREAD

**Issue by konotorii** [kelektiv/node-cron#657]
### Description

I've had no problem in the 1+ year I've been doing cron jobs using @nestjs/schedule which uses node-cron.

Now all of a sudden I'm getting the following error:

```
F:\Projects\backend\node_modules\cron\lib\time.js:271
                                                        Time Zone: ${zone || '""'} - Cron String: ${this} - UTC offset: ${date.format(
                                                                                                                               ^

TypeError: date.format is not a function
```

I do not understand why this could happen as I didn't update any packages recently that may have broken it. I tried updating and going back a few version to see if the issue would be resolved but was met with the same error.

**Comment by jroa919** [comment]
> ### Description
> I've had no problem in the 1+ year I've been doing cron jobs using @nestjs/schedule which uses node-cron.
> 
> Now all of a sudden I'm getting the following error:
> 
> ```
> F:\Projects\backend\node_modules\cron\lib\time.js:271
>                                                         Time Zone: ${zone || '""'} - Cron String: ${this} - UTC offset: ${date.format(
>                                                                                                                                ^
> 
> TypeError: date.format is not a function
> ```
> 
> I do not understand why this could happen as I didn't update any packages recently that may have broken it. I tried updating and going back a few version to see if the issue would be resolved but was met with the same error.
> 
> ### Screenshots
> _No response_
> 
> ### Additional information
> _No response_

Experiencing the same issue here. A deployed Nest.js backend shutdown and became unresponsive due to this error and led me here. A job configured using `CronExpression.EVERY_DAY_AT_1AM` kept failing to determine the next run date and timing out triggered this error.

**Comment by konotorii** [comment]
I was able to temporarily fix this by downgrading to:

"@nestjs/schedule": "2.1.0"
"cron": "2.1.0"
"luxon": "3.2.1"

Hope this can be a fix until root cause is found 👍

**Comment by intcreator** [comment]
the `date.format` fix will be out with 2.3.0, just dotting all my i's and crossing my t's before I personally publish this to NPM for the first time. however it looks like there was a larger issue that triggered the failsafe (see #467) which often has to do with the CPU being overutilized, infinite loops, etc.

**Comment by intcreator** [comment]
published now
